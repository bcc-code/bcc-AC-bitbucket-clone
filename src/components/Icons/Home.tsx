import * as React from "react"
import { IIconProps } from "@/types"

const Home: React.FC<IIconProps> = ({ customSize, className }) => {
  let customeClassName = className ? className : `w-6 h-6`
  return (
    <svg
      className={`${customeClassName} stroke-current hover:fill-current`}
      viewBox="0 0 22 18"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M1 7.66667L3 6.33333M21 7.66667L19 6.33333M19 6.33333L11 1L3 6.33333M19 6.33333V17H14V13C14 11 12.3125 10 11 10C9.6875 10 8 11 8 13V17H3V6.33333" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" />

    </svg>
  )
}

export default Home
