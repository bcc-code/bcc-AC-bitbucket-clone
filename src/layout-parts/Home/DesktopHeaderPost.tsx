import * as React from 'react';
import Link from '@/components/CustomLink'
import LazysizesFeaturedImage from '@/components/Images/LazysizesImage'
import VideoHeader from '@/components/Images/Video16to9'
import { IPostItem } from '@/types'
import PostMeta from '@/components/PostMeta/PostMeta'
import Bookmark from '@/components/PostElements/ToggleBookmark'
import TextSizeWClamp from '@/components/PostElements/TextSizeWClamp'

import { ReadingTimingAuthor, PostLabel } from '@/components/PostElements'
const rbgToString = (colors: number[], alpha?: number) => {
    if (alpha) {
        return "rgba(" + colors.join(", ") + ", " + alpha + ")";
    } else {
        return "rgb(" + colors.join(", ") + ")";
    }
}

const HeaderPost: React.FC<IPostItem> = ({ format, duration, image, title, excerpt, authors, reading_time, id, slug, media }) => {
    /* const {  muted } = palette; */
    const bgGradientStyle = { 'backgroundImage': '' }


    return (
        <div>
            <div
                className="absolute inset-x-0 top-0 h-64 z-minus"
            >
            </div>
            <div className="z-10 grid grid-cols-8 gap-12 standard-max-w-px relative pt-8 sm:pt-16 md:pt-20 lg:pt-24">
                <Link to={`/${slug}`} className="col-start-1 col-end-6 relative">


                    {media && media.video && media.video.src ? (
                        <VideoHeader
                            src={media.video.src}
                            className={`rounded-xxl sm:rounded-xl overflow-hidden`}
                        />

                    ) : (
                            <div className="m-0 relative w-full pb-square sm:pb-half">
                                <LazysizesFeaturedImage
                                    {...image}
                                    className="absolute w-full h-full inset-0 rounded-xxl sm:rounded-xl object-cover g-image"
                                />
                            </div>
                        )
                    }

                </Link>
                <div className="col-start-6 col-end-9  ml-5 flex flex-col justify-center pt-0">
                    <Link
                        to={`/${slug}`}

                    >
                        {format && format[0] && (
                            <PostLabel text={format[0].name} />
                        )}
                        <TextSizeWClamp
                            rawText={title}
                            bold="font-semibold"
                            fontKey="header-post"
                            clamp={3}
                            className="sm:py-3 py-1"

                        />
                    </Link>
                    <span className="font-medium hidden sm:block lg:text-xl mt-4">{excerpt}</span>
                    {authors && (
                        <span className="text-d4gray-dark lg:hidden sm:py-3 py-1 text-sm">
                            <PostMeta authors={authors} />
                        </span>
                    )}
                    <div className="flex items-center sm:py-3 py-1">
                        <span className="pr-2">
                            <Bookmark id={id} size={"6"} />
                        </span>
                        {reading_time && <p className="ml-1 text-sm text-d4gray-dark lg:hidden">{reading_time.text}</p>}
                        <ReadingTimingAuthor className="w-full text-sm text-d4gray-dark hidden lg:block" duration={duration?.listen} authors={authors} />
                    </div>
                </div>
            </div>
        </div>

    )
}

export default HeaderPost