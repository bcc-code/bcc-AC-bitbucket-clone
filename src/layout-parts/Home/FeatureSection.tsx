import React from 'react'
import FeaturedCard, { IFeaturedCard } from '@/components/PostItem/FeaturedCardNew'
import FeatureIcon from '@/components/Icons/Feature'
import MobileFeatureCard from '@/components/PostItem/WTopicFollowCard'
import ac_strings from '@/strings/ac_strings.json'

const FeatureSection: React.FC<{ featuredPosts: IFeaturedCard[] }> = ({ featuredPosts }) => {
    return (
        <div>
            <h3 className="relative mt-8 sm:mt-16 mx-4 mb-2 sm:mb-8 pb-2 text-d4dark text-base sm:border-b">
                <div className="flex items-center">
                    <FeatureIcon className="w-4 h-4" />
                    <span className="block mx-2">{ac_strings.featured}</span>
                </div>
            </h3>
            <div className="scroll-4col-h mx-4 my-4 hidden sm:grid">
                {featuredPosts.map((item, i) => {

                    return (
                        <div className={`div${i + 1}`} key={item.slug}>
                            <FeaturedCard className="bg-white" {...item} />
                        </div>
                    )
                })}
            </div>
            <div className="scroll-4col-h flex mb-4 sm:hidden">
                {featuredPosts.map((item, i) => {
                    const topic = item.topics && Array.isArray(item.topics) ? item.topics[0] : undefined
                    if (item.type) {
                        return (
                            <div className={`div${i + 1}`} key={item.slug}>
                                <FeaturedCard className="bg-white" {...item} />
                            </div>

                        )
                    } else {
                        return (
                            <div className={`div${i + 1}`} key={i}>
                                <MobileFeatureCard post={item} topic={topic} />
                            </div>
                        )
                    }

                })}
                <div className="min-w-4">

                </div>
            </div>
        </div>
    )
}

export default FeatureSection