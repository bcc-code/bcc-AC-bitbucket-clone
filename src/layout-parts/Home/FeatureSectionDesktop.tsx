import React from 'react'
import { FetchOnePost } from '@/HOC/FetchPosts'
import FeaturedCard, { IFeaturedCard } from '@/components/PostItemCards/FeaturedCard'

import TopImg from '@/components/PostItemCards/TopImg'
import ac_strings from '@/strings/ac_strings.js'
import { FetchLatestPlaylists, FetchLatestPodcast } from '@/HOC/FetchLatest'
import { playlistToPost, getRandomArray } from '@/helpers'
import { ITopicPostSlugs, IPostItem, ITopicPostItems } from '@/types'
import shortid from 'shortid'
const FeatureSection: React.FC<{ featuredPosts: IPostItem[] }> = ({ featuredPosts }) => {

    return (
        <div>
            <h3 className="relative mt-8 mx-4 mb-2 sm:mb-8 pb-2 text-d4dark text-base sm:border-b">
                <div className="flex items-center ">
                    <span className="block mx-2">{ac_strings.featured}</span>
                </div>
            </h3>
            <div className=" mx-4 my-4 grid gap-4 sm:gap-6 md:gap-6 grid-cols-4">
                <FetchLatestPodcast
                    key={shortid()}
                    layout="one"
                    render={({ podcastEps }) => <FeaturedCard  {...podcastEps[0]} type="podcast"
                    />}

                />
                <FetchLatestPlaylists
                    layout="one"
                    key={shortid()}
                    render={({ playlists }) => {
                        const random = getRandomArray(playlists, 1)
                        const post = random.length ? random[0] : undefined
                        return post ? (
                            <FeaturedCard {...playlistToPost(post)} type="playlist" />
                        ) : (<div></div>)
                    }}
                />
                <TopImg {...featuredPosts[0]} key={shortid()} />
                <TopImg {...featuredPosts[1]} key={shortid()} />

            </div>

        </div>
    )
}

export default FeatureSection