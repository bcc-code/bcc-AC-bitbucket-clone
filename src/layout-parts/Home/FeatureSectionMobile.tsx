import React from 'react'
import { FetchPostsFromSlugs } from '@/HOC/FetchPosts'

import RightImg from '@/components/PostItemCards/RightImg'
import { getRandomArray } from '@/helpers'
import { ITopicPostSlugs } from '@/types'
const FeatureSection: React.FC<{ topicPosts: ITopicPostSlugs[] }> = ({ topicPosts }) => {

    let postSlugs: string[] = []
    topicPosts.map(t => {
        postSlugs.push(...t.posts)
    })
    const randomFeaturedFromTopics = getRandomArray(postSlugs, 3)
    return (


        <FetchPostsFromSlugs
            slugs={randomFeaturedFromTopics}
            layout="list"
            render={({ posts }) => {
                return (
                    <div className="px-4">
                        {posts.map(item => {
                            return (
                                <RightImg {...item} />
                            )
                        })}
                    </div>
                )
            }
            }
        />


    )
}

export default FeatureSection