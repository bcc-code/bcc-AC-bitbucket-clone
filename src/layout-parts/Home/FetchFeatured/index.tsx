import * as React from "react"
import { IPostItem, } from '@/types'
import Placeholder from '@/layout-parts/Loader/MainpagePlaceholder'
import { fetchLocalPostsFromSlugs } from '@/helpers/fetchLocalData'
const FetchPosts: React.FC<{ slugs: string[], layout: "row" | "list", component: any }> = ({ children, slugs, component: Component }) => {
    const [posts, setPosts] = React.useState<IPostItem>([])
    const [loading, setLoading] = React.useState(false)
    React.useEffect(() => {
        setLoading(true)
        fetchLocalPostsFromSlugs(slugs)
            .then(res => {
                if (res) {
                    setLoading(false)
                    setPosts(res)
                }
            })
    }, slugs)
    return (
        <Placeholder loading={loading}>
            <Component posts={posts} />
        </Placeholder >
    )

}