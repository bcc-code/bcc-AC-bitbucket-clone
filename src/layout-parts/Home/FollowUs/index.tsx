import * as React from "react"
import SocialPlatforms from '@/layout-parts/Nav/SocialPlatforms'
import ac_strings from '@/strings/ac_strings'
const FollowUs = () => {
    return (
        <div className="px-4 rounded-lg flex flex-col" style={{ backgroundImage: 'linear-gradient(#edf2f7,#fff)' }}>
            <h6 className="text-d4slate-dark text-lg font-bold py-8 border-b">{ac_strings.follow_us}</h6>
            <SocialPlatforms col />
        </div>
    )
}

export default FollowUs