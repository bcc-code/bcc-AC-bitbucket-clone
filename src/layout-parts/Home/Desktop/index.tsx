import * as React from 'react'
import { DesktopFeaturedPostLoader } from '@/layout-parts/Loader/PlaceHolders'
import LatestSectionHeader from '@/layout-parts/LatestSectionHeader'
import LatestSection from '@/layout-parts/List/PostRow4Col'
import FeatureSectionDesktop from './FeatureSectionDesktop'
import HomeTopFeaturePost from '@/layout-parts/Home/HeaderPost'
import LowerSections from './LowerSections'
import ShowMore from '@/layout-parts/ShowMorePosts'
import RightImgWDes from '@/components/PostItemCards/RightImg'
import shortid from 'shortid'
import { IPostItem, ITopicPostItems } from '@/types'
import ac_strings from '@/strings/ac_strings.js'
interface IHomeMobileProps {
    mixed: IPostItem[]
    latest: IPostItem[]
    popular: IPostItem[]
    popularTopicsAll: {
        static: ITopicPostItems[]
        dynamic: ITopicPostItems[]
    }
}
const HomeDesktop: React.FC<IHomeMobileProps> = ({ mixed, latest, popular, popularTopicsAll }) => {
    const latestPostAsTopic = {
        id: '',
        name: ac_strings.latest,
        slug: ac_strings.slug_latest
    }
    return (

        <div className="hiddem sm:block">
            <DesktopFeaturedPostLoader loading={typeof mixed[0] === "undefined"}>
                <HomeTopFeaturePost {...mixed[0]} key={shortid()} />
            </DesktopFeaturedPostLoader>

            <div className="px-4">
                <LatestSectionHeader latestSlug={latestPostAsTopic.slug} />
                <LatestSection posts={latest.slice(0, 4)} />
                <FeatureSectionDesktop
                    featuredPosts={mixed.slice(2)}
                />
                <LowerSections
                    lists={popularTopicsAll.static}
                    newPostsForYou={[]}
                    topicsForYou={popularTopicsAll.static}
                    popularPosts={popular}
                />
                <div className="grid grid-cols-4 gap-4 md:gap-6 sm:px-4">
                    <div className="col-start-1 col-end-3 lg:col-end-4">
                        {latest.slice(6, 12).map((item, i) => {
                            return (
                                <div className={`mt-6 sm:mt-8 mx-4 sm:mr-10 sm:ml-0 div-post`} key={shortid()}>
                                    <RightImgWDes key={i} {...item} />
                                </div>
                            )
                        })}
                        <ShowMore
                            slug={latestPostAsTopic.slug}
                            startNr={2}
                        />
                    </div>
                </div>
            </div>
        </div>
    )
}

export default HomeDesktop