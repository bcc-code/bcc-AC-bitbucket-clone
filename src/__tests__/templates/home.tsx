import React from 'react';
import renderer from 'react-test-renderer';
import Home, { IHomeProps } from '@/templates/page/home'
import { props } from './homeProps'
it('renders correctly', () => {

    const tree = renderer
        .create(
            <Home {...props} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
})


/* const homeProps:IHomeProps =  */