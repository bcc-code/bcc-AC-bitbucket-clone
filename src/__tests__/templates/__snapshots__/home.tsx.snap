// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/www.activechristianity.org"
  onClick={[Function]}
  onMouseEnter={[Function]}
/>
`;
